src/org/apache/xerces/parsers/AbstractDOMParser.java

```diff
@@ -933,6 +942,11 @@ public void startElement (QName element, XMLAttributes attributes, Augmentations
                 return;
             }
             Element el = createElementNode (element);
+            // Extract location info if feature is enabled
+            if (fIncludeLocationInfo) {
+                DOMLocator elemLoc = new DOMLocatorImpl(fLocator.getLineNumber(), fLocator.getColumnNumber(), null);
```

Comment:
So the parse already provides this fLocator which is itself a DOMLocator, right? is it mutable? if not, we can just reuse it.

If it is mutable, can we copy any other fields over to the new DOMLocator?

Reply:
fLocator is of type XMLLocator not DOMLocator, and its state gets mixed into other objects and it is mutated, so we have to copy out of it. 

However, there is other info we can copy which is appropriate for a DOMLocator, and DOMLocators reference their node, so I changed this to copy as much as possible and provide the related node arg also.